packages/react-query/src/__tests__/useQuery.test.tsx

```diff
@@ -2231,7 +2228,28 @@ describe('useQuery', () => {
     // 7. Observer options updated
     // 8. Observer result updated -> stale
     // 9. Observer options updated
-    expect(fn).toHaveBeenCalledTimes(9)
```

Comment:
the comments about the "9 updates" is still here, but the assertion is gone 🤔 

I also don't know what the `states.push` are actually doing here. Nothing it seems.

I think this is one of the tests that should actually live in the query-core, not react-query. There is no nothing react specific about `should notify query cache when a query becomes stale`

we could do:

- subscribe to queryCache
- create an observer with a staleTime
- do the same checks there ...

this would also get rid of the test duplication in solidJs

Reply:
All right, I'll give that a try